include/picongpu/fields/MaxwellSolver/Yee/YeePML.hpp

```diff
@@ -0,0 +1,192 @@
+    {
+    private:
+
+        // Polynomial order of the absorber strength growth towards borders
+        // (often denoted 'n' in the literature)
+        uint32_t pmlOrder;
+
+        // These parameters are specific for each side,
+        // as a temporary solution sides are encoded in the absorber format (to be fixed)
+        uint32_t thickness[3][2]; // unit: cells
+        float_X maxAbsorbtion[3][2]; // often denoted 'sigma_max' in the literature
+
+        template<uint32_t AREA>
+        void updateE()
+        {
+            // In the border area apply PML when necessary
+            typedef SuperCellDescription<
+                    SuperCellSize,
+                    typename CurlB::LowerMargin,
+                    typename CurlB::UpperMargin
+                    > BlockArea;
+            using Difference = typename CurlB::Difference;
+
+            AreaMapping<AREA, MappingDesc> mapper(m_cellDescription);
+
+            constexpr uint32_t numWorkers = pmacc::traits::GetNumWorkers<
+                pmacc::math::CT::volume< SuperCellSize >::type::value
+            >::value;
+
+            auto splitFieldE = this->fieldB->getDeviceDataBox(); /// placeholder for now
+            PMACC_KERNEL(yeePML::KernelUpdateE< numWorkers, BlockArea >{ })
+                ( mapper.getGridDim(), numWorkers )(
+                    CurlB( ),
+                    this->fieldE->getDeviceDataBox(),
+                    splitFieldE,
+                    this->fieldB->getDeviceDataBox(),
+                    mapper
+                );
+        }
+
+        template<uint32_t AREA>
+        void updateBHalf()
+        {
+            typedef SuperCellDescription<
+                    SuperCellSize,
+                    typename CurlE::LowerMargin,
+                    typename CurlE::UpperMargin
+                    > BlockArea;
+
+            AreaMapping<AREA, MappingDesc> mapper(m_cellDescription);
+
+            constexpr uint32_t numWorkers = pmacc::traits::GetNumWorkers<
+                pmacc::math::CT::volume< SuperCellSize >::type::value
+            >::value;
+
+            auto splitFieldB = this->fieldB->getDeviceDataBox(); /// placeholder for now
+            PMACC_KERNEL(yeePML::KernelUpdateBHalf< numWorkers, BlockArea >{ })
+                (mapper.getGridDim(), numWorkers)(
+                    CurlE(),
+                    this->fieldB->getDeviceDataBox(),
+                    splitFieldB,
+                    this->fieldE->getDeviceDataBox(),
+                    mapper
+                );
+        }
+
+    public:
+
+        YeePML(MappingDesc cellDescription) : Yee(cellDescription)
+        {
+            // for now reuse parameters of the old absorber
+            // WARNING: this is only correct for a single process,
+            // TODO: for the general case needs to take into account domain decomposition
+            // so that thickness is 0 for "internal" boundaries between domains
```

Comment:
note: The thickness must be updated within each time step if the moving window is enabled. The reason is that our devices have no fixed position within the simulation volume (https://github.com/ComputationalRadiationPhysics/picongpu/wiki/PIConGPU-domain-definitions#moving-window)